pygatt/bled112_backend.py

```diff
@@ -385,28 +365,18 @@ def connect(self, address, timeout=5,
         timeout -- number of seconds to wait before returning if not connected.
         addr_type -- one of the ble_address_type constants.
 
-        Returns True if the connection was completed successfully.
-        Returns False otherwise.
+        Raises BLED112Error on failure.
         """
-        # Get locks
-        self._get_locks()
-
         # Make sure there is NOT a connection
         if self._connected:
-            self._logger.warn("Not connected")
-            self._loglock.release()
-            self._main_thread_cond.release()
-            return False
-
-        # Setup connection timeout timer
-        self._connect_timeout = False
-        timer = threading.Timer(timeout, self._timer_func)
+            self._logger.warn("Already connected")
+            raise BLED112Error("Tried to connect when already connected")
 
         # Connect to the device
         bd_addr = [b for b in address]
         interval_min = 6  # 6/1.25 ms
         interval_max = 30  # 30/1.25 ms
-        timeout = 20  # 20/10 ms
+        supervision_timeout = 20  # 20/10 ms
```

Comment:
These values could use some context.  Also, make sure that the unit is apparent.  It's Stratos tradition to maintain a healthy fear of https://en.wikipedia.org/wiki/Mars_Climate_Orbiter#Cause_of_failure